test/intersections.jl

```diff
@@ -103,6 +108,139 @@
     s2 = Segment(P2(0.94798386, 0.5344541), P2(0.9472896, 0.5340202))
     @test s1 ∩ s2 ≈ s2 ∩ s1 ≈ P2(0.94798386, 0.5344541)
 
+    s₁ = Segment(P2(0,0), P2(3,4))
+    s₂ = Segment(P2(1,2), P2(3,-2))
+    s₃ = Segment(P2(2,0), P2(-2,0))
+    s₄ = Segment(P2(0,0), P2(1,2))
+    s₅ = Segment(P2(1,2), P2(3,4))
+    s₆ = Segment(P2(-1,-4/3), P2(0,0))
+    s₇ = Segment(P2(1,2), P2(0,4))
+    s₈ = Segment(P2(4,16/3), P2(3,4))
```

Comment:
Notice that the code style you adopted in this section is different than the surrounding code style. We can ignore it this time because these are just tests, but it is important to try to reuse the surrounding code style.